src/detect-smb-share.c

```diff
@@ -143,15 +151,19 @@ void DetectSmbShareRegister(void)
     sigmatch_table[KEYWORD_ID].name = KEYWORD_NAME;
     sigmatch_table[KEYWORD_ID].alias = KEYWORD_NAME_LEGACY;
     sigmatch_table[KEYWORD_ID].Setup = DetectSmbShareSetup;
-    sigmatch_table[KEYWORD_ID].flags |= SIGMATCH_NOOPT|SIGMATCH_INFO_STICKY_BUFFER;
+    sigmatch_table[KEYWORD_ID].flags |= SIGMATCH_NOOPT | SIGMATCH_INFO_STICKY_BUFFER;
     sigmatch_table[KEYWORD_ID].desc = "sticky buffer to match on SMB share name in tree connect";
 
-    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOSERVER, 2,
-            PrefilterGenericMpmRegister, GetShareData,
-            ALPROTO_SMB, 1);
+    sigmatch_table[KEYWORD_ID_LEGACY].name = KEYWORD_NAME_LEGACY;
```

Comment:
I don't see the value of this. We already have `alias` in the original code. We can use that for output and such without growing the sigmatch_table

Reply:
Please correct me if I'm wrong but alias itself won't be enough since when I run `list-keywords` it will call `SigTableList` which would just use those keywords with a `.name` to be printed and not when passed with the `.alias` name.

So for example:
```
--list-keywords=tls_cert_subject 
Non existing keyword
```

Even though it's an alias for `tls.cert_subject`. Maybe we want to print `.alias` as well with the `--list-keywords` and reference to the actual `name`? That would avoid several changes on the protocol code where I would argue we would have to add the old/alternative names to the table with `.name`.